vita3k/shader/src/usse_utilities.cpp

```diff
@@ -678,13 +678,13 @@ spv::Id shader::usse::utils::unpack_one(spv::Builder &b, SpirvUtilFunctions &uti
     case DataType::UINT8:
     case DataType::UINT16:
     case DataType::INT16: {
-        if (utils.unpack_funcs.find(type) == utils.unpack_funcs.end()) {
+        if (!utils.unpack_funcs.contains(type)) {
             utils.unpack_funcs[type] = make_unpack_func(b, features, type);
         }
         return b.createFunctionCall(utils.unpack_funcs.at(type), { scalar });
```

Comment:
```suggestion
        auto iter = utils.unpack_funcs.find(type);
        if (iter == utils.unpack_funcs.end()) {
            iter = utils.unpack_funcs.emplace(type,make_unpack_func(b, features, type)).first;
        }
        return b.createFunctionCall(iter->second, { scalar });
```
same suggestion for two next changes

Reply:
it's not critical, corresponding to https://github.com/Vita3K/Vita3K/pull/2913#pullrequestreview-1622179494

Reply:
I used `try_emplace` instead

Reply:
`try_emplace` run `make_pack_func(b, features, source_type)` every time and then throw away its result.

Reply:
Are you sure? :thinking: 
From https://rules.sonarsource.com/cpp/RSPEC-6030/:
> C++17 introduced try_emplace that does not construct the value if the key is already present in the map and constructs the value in place if necessary.

Reply:
arguments of try_emplace (and any other function) are evaluated before function call

Reply:
just log calls of `make_*_func` and check it yourself

Reply:
My bad, fixed. Thanks for the review 👍